source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -182,8 +182,11 @@ void MaterialXNode::createAndRegisterFragment()
 
 void MaterialXNode::reloadDocument()
 {
-    _document = nullptr;
+    _document.reset();
+    _materialXData.reset();
```

Comment:
If we don't reset it here and it was valid before, it will keep the old MaterialXData alive until the shared pointer is reset to the new one, and MaterialXData will keep the document alive as well. So, two instances of the data would exist in memory simultaneously with no benefit. Also, if `createAndRegisterFragment()` fails afterwards for any reason (e.g. the file has been deleted), the old MaterialXData will live on and the old material will remain in the viewport after the refresh, giving the user the erroneous impression that the reload has successfully finished and the material is up to date and valid.